### ISSUE TITLE
`ruff` disregards per file ignore on `E402`

### ISSUE THREAD

**Issue by niccolomineo** [astral-sh/ruff#9246]
For some reason the following directive is ignored:
```
"myproject/*sgi.py" = [
    "E402",
]
```
Issue:
```
$ ruff . --fix
myproject/wsgi.py:15:1: E402 Module level import not at top of file
Found 1 error.
$ ruff --version
ruff 0.1.9
```

Relevant settings:

```
[tool.ruff]
ignore = [
    "D203",
    "D212",
]
select = [
    "A",      # flake8-builtins
    "B",      # flake8-bugbear
    "C4",     # flake8-comprehensions
    "C90",    # McCabe cyclomatic complexity
    "D",      # pydocstyle
    "DJ",     # flake8-django
    "E",      # pycodestyle errors
    "F",      # Pyflakes
    "I",      # isort
    "Q",      # flake8-quotes
    "UP",     # pyupgrade
    "W",      # pycodestyle warnings
]
target-version = "py312"

[tool.ruff.per-file-ignores]
"myproject/*sgi.py" = [
    "E402",
]
"*/migrations/*.py" = [
    "D100",
    "D101",
    "D102",
    "D104",
]

[tool.ruff.isort]
known-first-party = [
    "myproject",
]
known-third-party = [
    "django",
]
```

**Comment by charliermarsh** [comment]
@niccolomineo - It looks like you're using `myproject` in the configuration file, but the directory on disk is called `project` ("project/wsgi.py:15:1: E402 Module level import not at top of file"). Could that be the culprit?

**Comment by niccolomineo** [comment]
> @niccolomineo - It looks like you're using `myproject` in the configuration file, but the directory on disk is called `project` ("project/wsgi.py:15:1: E402 Module level import not at top of file"). Could that be the culprit?

Sorry, double checked and it's my bad (I just failed at manually obfuscating the real dir name).

**Comment by niccolomineo** [comment]
The issue unfortunately still stands.

**Comment by niccolomineo** [comment]
@charliermarsh I see what the problem is. Those rules I reported in the opening post are in my `pyproject.toml`, but I also added a `ruff.toml` with the sole following rule:

```
[format]
docstring-code-format = true
```

I guess `ruff.toml` is overriding `pyproject.toml`. Do I need to move all `ruff` related rules currently in `pyproject.toml` to `ruff.toml`? I'd prefer to move the `docstring-code-format = true` rule only. If this is possible, what's the right syntax?
